wallet/pskt/src/wasm/pskt.rs

```diff
@@ -384,4 +396,317 @@ impl PSKT {
+            State::Constructor(pskt) => pskt.signer().finalizer(),
+            State::Updater(pskt) => pskt.signer().finalizer(),
+            State::Signer(pskt) => pskt.finalizer(),
+            State::Combiner(pskt) => pskt.signer().finalizer(),
+            State::Finalizer(pskt) => pskt,
+            state => return Err(Error::state(state))?,
+        };
+
+        // pre-conditions check
+        for input in pskt_finalizer.inputs.iter() {
+            if input.redeem_script.is_some() {
+                return Err(Error::custom("finalize_p2pk does not support redeem scripts"));
+            }
+
+            if let Some(utxo_entry) = &input.utxo_entry {
+                let script_class = ScriptClass::from_script(&utxo_entry.script_public_key);
+
+                // only allow p2pk script class
+                match script_class {
+                    ScriptClass::NonStandard => return Err(Error::InvalidScriptClassError(ScriptClass::PubKey, script_class)),
+                    ScriptClass::PubKeyECDSA => return Err(Error::InvalidScriptClassError(ScriptClass::PubKey, script_class)),
+                    ScriptClass::ScriptHash => return Err(Error::InvalidScriptClassError(ScriptClass::PubKey, script_class)),
+                    ScriptClass::PubKey => (),
+                }
+            } else {
+                return Err(Error::custom("Input without UTXO entry cannot be finalized"));
+            }
+        }
+
+        let result = pskt_finalizer.finalize_sync(|inner: &Inner| -> std::result::Result<Vec<Vec<u8>>, String> {
+            inner
+                .inputs
+                .iter()
+                .map(|input| {
+                    if input.partial_sigs.len() != 1 {
+                        return Err(format!(
+                            "finalize_p2pk error: input for outpoint {} must have exactly one signature, but has {}",
+                            input.previous_outpoint,
+                            input.partial_sigs.len()
+                        ));
+                    }
+
+                    let signature_script: Vec<u8> = input
+                        .partial_sigs
+                        .values()
+                        .next()
+                        .map(|signature| {
+                            iter::once(OpData65).chain(signature.into_bytes()).chain([input.sighash_type.to_u8()]).collect()
+                        })
+                        .unwrap();
+
+                    Ok(signature_script)
+                })
+                .collect()
+        });
+
+        let finalized_pskt = match result {
+            Ok(finalized_pskt) => finalized_pskt,
+            Err(e) => return Err(Error::from(e.to_string())),
+        };
+
+        let mutable_transaction = finalized_pskt
+            .extractor()?
+            .extract_tx(&network_type.into())
+            .map_err(|e| Error::custom(format!("Failed to extract transaction: {e}")))?;
+
+        Ok(mutable_transaction.tx.into())
```

Comment:
finalizer can only build tx id and signature scripts.

extractor may perform simulation(by default) and calculates mass, also metable tx has tx, mass and utxo entries. pskt interface returns all of them to caller. if you assume that caller doesnt need them - u can extract and return tx as you r doing now